tests/fixtures.py

```diff
@@ -269,6 +283,114 @@ def inner(module_path, inventory_paths, basedir=None, tf_var_files=None,
   return inner
 
 
+def e2e_validator(module_path: str, extra_files: list, tf_var_files: list,
+                  basedir: os.PathLike = None):
+  """Function running apply, plan and destroy to verify the case end to end
+
+  1. Tests whether apply does not return errors
+  2. Tests whether plan after apply is empty
+  3. Tests whether destroy does not return errors
+  """
+  module_path = _REPO_ROOT / module_path
+  with _prepare_root_module(module_path) as test_path:
+    binary = os.environ.get('TERRAFORM', 'terraform')
+    tf = tftest.TerraformTest(test_path, binary=binary)
+    extra_files = [(module_path / filename).resolve()
+                   for x in extra_files or []
+                   for filename in glob.glob(x, root_dir=module_path)]
+    tf.setup(extra_files=extra_files, upgrade=True)
+    tf_var_files = [(basedir / x).resolve() for x in tf_var_files or []]
+
+    try:
+      apply = tf.apply(
+          tf_var_file=tf_var_files)
+      plan = tf.plan(
+          output=True,
+          tf_var_file=tf_var_files)
+      changes = dict((k, v['change'])
+                     for k, v in plan.resource_changes.items()
+                     if v.get('change', {}).get('actions') != ['no-op'])
+      # compare before with after to raise more meaningful failure to the user, i.e one
+      # that shows how resource will change
+      assert dict((k, v['before']) for k, v in changes.items()) == dict(
```

Comment:
can you make these more readable? using fewer lines usually is a good trick to hurt legibility